### PR TITLE
fs/procfs: refine file backtrace

### PR DESCRIPTION
**Pull request by anchao**
## Summary

fs/procfs: refine file backtrace

remove backtrace arrary in stack

## Impact
N/A
## Testing
ci-check

### CODE REVIEW

**Comment by xiaoxiang781216** [comment]
@XuNeo will provide a better fix, let's wait a moment.

**Comment by anchao** [comment]
@xiaoxiang781216 @XuNeo Please review the code internally before send PR to the community. https://github.com/apache/nuttx/pull/12532 looks like a workaround with too many basic issues

**Comment by anchao** [comment]
> please review this patch: #12585

@xiaoxiang781216 #12585 did not resolve the issue in this PR
1. Backtrace does not need to format twice
  a. format to temp buffer
  b. temp buffer to user buffer
->
 a. format to user buffer directly
3. remove backtrace variables defined to save the stack size

File: fs/procfs/fs_procfsproc.c [hunk 1]
```diff
@@ -1306,22 +1334,14 @@ static ssize_t proc_groupfd(FAR struct proc_file_s *procfile,
...
                                    i, filep->f_oflags,
                                    INODE_GET_TYPE(filep->f_inode),
-                                   (long)filep->f_pos, path,
-                                   file_dump_backtrace(filep,
-                                                       backtrace,
-                                                       sizeof(backtrace)
-                                                      )
-                                  );
+                                   (long)filep->f_pos, path);
+      linesize   = procfs_format_backtrace(procfile->line, STATUS_LINELEN,
```
Comment (xiaoxiang781216) [inline fs/procfs/fs_procfsproc.c:1341]: do you need procfile->line + linesize

File: fs/procfs/fs_procfsproc.c [hunk 2]
```diff
@@ -1306,22 +1334,14 @@ static ssize_t proc_groupfd(FAR struct proc_file_s *procfile,
...
+                                           linesize, filep);
       copysize   = procfs_memcpy(procfile->line, linesize,
                                  buffer, remaining, &offset);
-      if (linesize + 1 == STATUS_LINELEN)
-        {
-          procfile->line[STATUS_LINELEN - 2] = '\n';
-          linesize = STATUS_LINELEN;
-        }
```
[the remark this answers is not in the document]
Comment (anchao) [reply fs/procfs/fs_procfsproc.c:1324]: 1. profile->line is input, buffer in output, right? why need make sure input buffer with '\n' after memcpy?
2. procfs will copy multi times, Update line breaks are unnecessary

**Pull request closed without merging**